teps/0091-trusted-resources.md

```diff
@@ -241,47 +237,55 @@ The verification should be done in the Tekton Pipeline's reconciler after remote
+You can create multiple `VerificationPolicy` and apply them to the cluster.
+1. Trusted resources will look up policies from the resource namespace (this is fetched from taskrun/pipelinerun namespace).
+2. If multiple policies are found. For each policy we will check if the resource url is matching any of the `patterns` in the `resources` list. If matched then this policy will be used for verification.
+3. If multiple policies are matched, the resource needs to pass all of them to pass verification.
+4. To pass one policy, the resource must successfully match at least one public key in the policy.
 
 Example of the Key CRD:
 ```yaml
 apiVersion: tekton.dev/v1alpha1
-Kind: VerificationPolicy
+kind: VerificationPolicy
 metadata:
- name: verification-policy
- namespace: tekton-pipelines
+  name: verification-policy-a
+  namespace: resource-namespace
 spec:
-   resources:
-    - https://github.com/tektoncd/catalog.git
-    - gcr.io/tekton-releases/catalog/upstream/*
-   authorities:
-    - name: cosign
-      key: /etc/signing-secrets/cosign.pub
-    - name: kms
-      key: gcpkms://projects/<project>/locations/<location>/keyRings/<keyring>/cryptoKeys/<key>
-    - name: keyless
-      keyless:
-        url: "https://fulcio.sigstore.dev"
+  # resources defines a list of patterns
+  resources:
+    - pattern: "https://github.com/tektoncd/catalog.git"  #git resource pattern
+    - pattern: "gcr.io/tekton-releases/catalog/upstream/git-clone"  # bundle resource pattern
+    - pattern: " https://artifacthub.io/"  # hub resource pattern
+  # authorities defines a list of public keys
+  authorities:
+    - name: SecretKey
+      key:
+        # secretRef refers to a secret in the cluster, this secret should contain public keys data
+        secretRef:
+          name: secret-name-a
+          namespace: secret-namespace
+        hashAlgorithm: sha256
+    - name: InlineKey
+      key:
+        # data stores the inline public key data
+        data: "STRING_ENCODED_PUBLIC_KEY"
 ```
 
+`namespace` should be the same of corresponding resources' namespace.
+
+`pattern` is used to filter out remote resources by their sources URL. e.g. git resources pattern can be set to https://github.com/tektoncd/catalog.git. The `pattern` should follow regex schema, we use go regex library's [`Match`](https://pkg.go.dev/regexp#Match) to match the pattern from VerificationPolicy to the `ConfigSource` URL resolved by remote resolution. Note that `.*` will match all resources.
+To learn more about regex syntax please refer to [syntax](https://pkg.go.dev/regexp/syntax).
+To learn more about `ConfigSource` please refer to resolvers doc for more context. e.g. [gitresolver](https://github.com/tektoncd/pipeline/blob/main/docs/git-resolver.md#resolutionrequest-status)
```

Comment:
I have talked to @chuangw6 and we don't have a `ConfigSource` in our repo, so I linked to a resolver using `ConfigSource`. I will clarify this in the tep

Reply:
I think it might be worth briefly mentioning what is configsource first, and then where this field came from (i.e. ResolutionRequest status) and what subfield we use. 

https://github.com/tektoncd/chains/issues/521 has full details about how Chains captures this in provenance, but step 1-3 are pipeline-related work.

Reply:
I think https://github.com/tektoncd/pipeline/blob/main/docs/pipeline-api.md#configsource-1 might be a good link